src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.slice(-1)[0];
+  const toSeparator = toFormat.slice(-1)[0];
+  const dateElements = date.split(fromSeparator);
+
+  const dateMap = fromFormat.reduce((acc, format, index) => {
+    acc[format] = dateElements[index];
+
+    return acc;
+  }, {});
+
+  const convertYear = (year, format) => {
+    if (format === 'YYYY') {
+      return year.length === 2 ? (year < 30 ? '20' + year : '19' + year) : year;
+    }
+
+    if (format === 'YY') {
+      return year.slice(-2);
+    }
+
+    return year;
+  };
+
+  const formattedDateParts = toFormat.map((format) => {
+    if (format === 'YY' || format === 'YYYY') {
+      return convertYear(dateMap['YYYY'] || dateMap['YY'], format);
+    }
+
+    return dateMap[format];
+  });
+
+  return formattedDateParts.filter(Boolean).join(toSeparator);
```

Comment:
The mapping of `toFormat` assumes that the `dateMap` has all the necessary keys. This is a safe assumption given the task's constraints, but be cautious with this approach in a more general context, as it could lead to `undefined` values if the key does not exist in `dateMap`.